compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1358,6 +1456,28 @@ public BSymbol getBinaryEqualityForTypeSets(OperatorKind opKind, BType lhsType,
         return symTable.notFoundSymbol;
     }
 
+    public boolean markParameterizedType(BType type, BType constituentType) {
+        if (Symbols.isFlagOn(constituentType.flags, Flags.PARAMETERIZED)) {
+            type.tsymbol.flags |= Flags.PARAMETERIZED;
+            type.flags |= Flags.PARAMETERIZED;
+            return true;
+        }
+        return false;
+    }
+
+    public void markParameterizedType(BType enclosingType, Collection<BType> constituentTypes) {
```

Comment:
If the `enclosingType` already has `PARAMETERIZED` flag we don't need to set again. Can skip loop.

Reply:
The enclosing type would not contain the flag since this method gets called right after the enclosing type gets created.